### PR TITLE
Render sky before opaque pass in GLES3 so it never draws over objects

### LINKED ISSUE #29108: Objects with "No Depth Test" gets drawn behind the Sky. GLES3
**Godot version:**

3.1.1 stable

**OS/device including version:**

Ubuntu 18.04. RTX Quadro 5000, Driver Version: 418.67

**Issue description:**

Objects with "No Depth Test" gets drawn behind the sky. Can be avoided by either checking "Transparent" or setting the environment to a color.

https://youtu.be/YnBT_Qykjzk

**Steps to reproduce:**
1. Open a fresh project
2. Make two MeshInstances occluding each other.
2. Create SpatialMaterial, check "No Depth Test" flag
3. Set the appropriate render priority (1 for the object with the material).

### PR DESCRIPTION
**Pull request by clayjohn**
Fixes: https://github.com/godotengine/godot/issues/29108

This issue is not present in GLES2.

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -4447,6 +4447,17 @@ void RasterizerSceneGLES3::render_scene(const Transform &p_cam_transform, const
 		glDisable(GL_BLEND);
 	}
 
+	if (env && env->bg_mode == VS::ENV_BG_SKY && (!storage->frame.current_rt || (!storage->frame.current_rt->flags[RasterizerStorage::RENDER_TARGET_TRANSPARENT] && state.debug_draw != VS::VIEWPORT_DEBUG_DRAW_OVERDRAW))) {
+
+		/*
+		if (use_mrt) {
+			glBindFramebuffer(GL_FRAMEBUFFER,storage->frame.current_rt->buffers.fbo); //switch to alpha fbo for sky, only diffuse/ambient matters
```
Comment (aaronfranke) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: Should this comment still be here?

**Review by aaronfranke — APPROVED** [review]
I can confirm that this fixes the issue.

File: drivers/gles3/rasterizer_scene_gles3.cpp (hunk 1 above)
Comment (clayjohn) [reply drivers/gles3/rasterizer_scene_gles3.cpp]: My gut says no. But GLES3 isnt going to be around for long, so im trying to change as little as possible.

**Comment by clayjohn** [comment]
Superceded by https://github.com/godotengine/godot/pull/29132 which also fixes the bug, but still draws the sky after the opaque pass which is slightly faster.

**Pull request closed without merging**